pkg/rbac/rbac.go

```diff
@@ -211,21 +212,23 @@ func NewEnforceHandler(basePath string, enforcer *casbin.Enforcer) func(c echo.C
 		if !ok {
 			return false, errors.New("invalid URL")
 		}
-		switch resource {
-		case "namespaces":
-			// Always allow this operation to list namespaces,
-			// however, we filter the result based on permission.
-			return true, nil
-		default:
-			namespace := c.Param("namespace")
-			name := c.Param("name")
-			object = namespace + "/" + name
-		}
-
 		action, ok := actionMethodMap[c.Request().Method]
 		if !ok {
 			return false, errors.New("invalid method")
 		}
+		namespace := c.Param("namespace")
+		name := c.Param("name")
+		object = namespace + "/" + name
+		// Always allowing listing all namespaces.
+		// The result is filtered based on permission.
+		if resource == ResourceNamespaces {
+			return true, nil
+		}
+		// Always allow listing database engines.
+		// The result is filtered based on permission.
+		if resource == ResourceDatabaseEngines && name == "" && action == ActionRead {
```

Comment:
Should this be the following?
```suggestion
		if resource == ResourceDatabaseEngines && name == "" && (action == ActionRead || action == "*") {
```

Reply:
No, I don't think so..

The value of `action` is derived from the HTTP method, so it will always be one of `read`, `create`, `updated`... etc. Whereas `*` is a wildcard we set in the RBAC policy

Moreover, we want to bypass the handler only when Listing database-engines, i.e, `GET /v1/namespaces/database-engines`. For the other `database-engines` APIs, we still need this enforcer

Reply:
You're right, casbin handles that internally